lighthouse-core/audits/byte-efficiency/render-blocking-resources.js

```diff
@@ -272,6 +273,8 @@ class RenderBlockingResources extends Audit {
    * @return {Promise<LH.Audit.Product>}
    */
   static async audit(artifacts, context) {
+    if (artifacts.GatherContext.gatherMode !== 'navigation') return {score: 0, notApplicable: true};
```

Comment:
in an ideal world it seems like we wouldn't have to remember to do this either, but I don't know a way around it other than audit annotations (a la `meta.supportedModes`) and have `audit-runner` take care of the check, but I don't know if that's any better. Curious about your thoughts on it

Reply:
I like the idea of having it as an optinal dependency. I was imagining the supported modes would be inferred by the available artifacts. Maybe we could use `meta.applicableModes` or something to avoid mismatch.